src/runtime/HalideRuntimeHexagonHost.h

```diff
@@ -113,6 +113,16 @@ extern int halide_hexagon_set_performance_mode(void *user_context, halide_hexago
 extern int halide_hexagon_set_performance(void *user_context, halide_hexagon_power_t *perf);
 // @}
 
+/** Set the priority for Hexagon threads.
+ * - Valid priority values range from 1 to 255
+ * - Smaller number for higher priority
+ * - The highest priority for a user thread is 1, 0 reserved for OS usage
+ * - If not set, Halide thread priority will default to 100
+ * - This should be called before running a pipeline. */
```

Comment:
IMO this behavior is problematic. It's especially weird because the run_context thread will get a new priority, but the thread pool threads will not.

IIRC we started down this path of exploring priorities because we were debugging bad parallel performance issues. But, I don't think we ever actually concluded that thread priorities were an issue, correct?

I am wondering if we are adding some new API support/complexity for something that we don't actually have a use case for, and has some quirky behavior.

I know the design has been like this for a while, sorry for not saying this earlier.

Reply:
If you are talking about a case where the priority gets changed between pipeline dispatches, then yes, existing worker threads will not get the new priority.  If you like, we could avoid this behavior if we added logic to only allow a single call to set the thread priority....or a way to track down the existing worker threads and update their priorities.

This API was really intended to be used once at the beginning of an app.

The use case for this API actually came a while ago from a customer who had previously written Hexagon apps each with different priorities so they could all be running simultaneously and yet still provide the desired quality of service.  They wanted to be able to do the same thing when they used Halide.

Reply:
OK, in that case, we can go ahead with this, modulo some minor comments.